velox/common/base/tests/SimdUtilTest.cpp

```diff
@@ -349,4 +349,22 @@ TEST_F(SimdUtilTest, Batch64_memory) {
   EXPECT_EQ(data[1], 2);
 }
 
+template <typename T>
+void validateReinterpretBatch() {
+  using U = typename std::make_unsigned<T>::type;
+  auto result = simd::reinterpretBatch<U>(simd::setAll<T>(-1));
+  auto result2 = simd::reinterpretBatch<T>(simd::setAll<T>(-1));
+  for (int i = 0; i < result.size; ++i) {
+    EXPECT_EQ(result.get(i), std::numeric_limits<U>::max());
+    EXPECT_EQ(result2.get(i), -1);
+  }
+}
+
+TEST_F(SimdUtilTest, reinterpretBatch) {
+  validateReinterpretBatch<int8_t>();
```

Comment:
Can we add to test from signed to unsigned?
Also can we test conversions like int8 to int16?

Reply:
This test is for exact signed to unsigned.  Conversion between different width is not supported yet as no use case for that yet.